.github/workflows/ci.yml

```diff
@@ -46,92 +34,97 @@ jobs:
         with:
           version: ${{ env.MISE_VERSION }}
       - name: Install Tools
-        run: ${{ matrix.install }}
-      - name: Lint ${{ matrix.sdk }}
-        run: ${{ matrix.lint }}
+        run: |
+          go run ./cmd/dev install ${{ matrix.tools }}
+      - name: Add Go bin to PATH
+        run: |
+          echo "$(go env GOPATH)/bin" >> $GITHUB_PATH
+      - name: Check ${{ matrix.sdk }} worker
+        run: |
+          go run ./cmd/dev check ${{ matrix.sdk }}
+      - name: Test ${{ matrix.sdk }} worker
+        run: |
+          go run ./cmd/dev test ${{ matrix.sdk }}
       - name: Check formatting
         run: |
-          git diff --exit-code || (echo "${{ matrix.sdk }} worker has uncommitted formatting changes after linting" && exit 1)
+          git diff --exit-code || (echo "${{ matrix.sdk }} worker has uncommitted formatting changes" && exit 1)
 
   build-worker:
     runs-on: ubuntu-latest
     strategy:
       fail-fast: false
       matrix:
         sdk: [go, java, python, typescript, dotnet]
-        include:
-          - sdk: go
-            install: ./scripts/install-go.sh
-            build: ./scripts/build-go-worker.sh
-            submodules: false
-          - sdk: java
-            install: ./scripts/install-java.sh && ./scripts/install-go.sh
-            build: ./scripts/build-java-worker.sh
-            submodules: false
-          - sdk: python
-            install: ./scripts/install-python.sh && ./scripts/install-go.sh
-            build: ./scripts/build-python-worker.sh
-            submodules: false
-          - sdk: typescript
-            install: ./scripts/install-node.sh && ./scripts/install-go.sh
-            build: ./scripts/build-typescript-worker.sh
-            submodules: true
-          - sdk: dotnet
-            install: ./scripts/install-dotnet.sh && ./scripts/install-go.sh
-            build: ./scripts/build-dotnet-worker.sh
-            submodules: true
     steps:
       - name: Checkout repo
         uses: actions/checkout@v4
         with:
-          submodules: ${{ matrix.submodules }}
-      - name: Install mise
-        uses: jdx/mise-action@v2.4.4
-        with:
-          version: ${{ env.MISE_VERSION }}
-      - name: Install Tools
-        run: ${{ matrix.install }}
-      - name: Build ${{ matrix.sdk }} worker
-        run: ${{ matrix.build }}
+          submodules: true
+      - name: Load versions
+        run: |
+          set -a && source versions.env && set +a
+          printenv >> $GITHUB_ENV
+      - name: Set SDK version variable
+        run: |
+          SDK_VAR_NAME="${{ matrix.sdk }}_SDK_VERSION"
+          SDK_VAR_NAME=$(echo "$SDK_VAR_NAME" | tr '[:lower:]' '[:upper:]')
+          echo "SDK_VERSION=${!SDK_VAR_NAME}" >> $GITHUB_ENV
+      - name: Build ${{ matrix.sdk }} worker image
+        run: |
+          go run ./cmd build-worker-image \
+            --language ${{ matrix.sdk }} \
+            --version v${{ env.SDK_VERSION }} \
```

Comment:
For example, if `--version` isn't given, we can default to loading the version from the file, or we can take `--versions-file` as an argument.